runelite-client/src/main/java/net/runelite/client/plugins/hydra/HydraPlugin.java

```diff
@@ -0,0 +1,424 @@
+import net.runelite.api.events.NpcSpawned;
+import net.runelite.api.events.ProjectileMoved;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.events.ConfigChanged;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.PluginType;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+@PluginDescriptor(
+	name = "Hydra Helper",
+	description = "Overlays for normal Hydras.",
+	tags = {"hydra", "helper", "baby", "small", "normal", "regular"},
+	type = PluginType.PVM,
+	enabledByDefault = false
+)
+@Slf4j
+@Singleton
+public class HydraPlugin extends Plugin
+{
+	private static final Set<HydraAnimation> VALID_HYDRA_ANIMATIONS = Set.of(
+		HydraAnimation.RANGE,
+		HydraAnimation.MAGIC
+	);
+
+	@Getter(AccessLevel.PACKAGE)
+	private Map<Integer, Hydra> hydras = new HashMap<>();
+
+	@Getter(AccessLevel.PACKAGE)
+	private Map<LocalPoint, Projectile> poisonProjectiles = new HashMap<>();
+
+	@Getter(AccessLevel.PACKAGE)
+	private NPC interactingNpc = null;
+
+	private boolean attackCounterOverlay = false;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean boldAttackCounterOverlay = false;
+
+	private boolean prayerOverlay = false;
+
+	private boolean poisonOverlay = false;
+
+	@Getter(AccessLevel.PACKAGE)
+	@Inject
+	private Client client;
+
+	@Inject
+	private HydraConfig hydraConfig;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private HydraAttackCounterOverlay hydraAttackCounterOverlay;
+
+	@Inject
+	private HydraPrayerOverlay hydraPrayerOverlay;
+
+	@Inject
+	private HydraPrayerAttackCounterOverlay hydraPrayerAttackCounterOverlay;
+
+	@Inject
+	private HydraPoisonOverlay hydraPoisonOverlay;
+
+	@Provides
+	HydraConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(HydraConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		updateConfig();
+
+		if (attackCounterOverlay)
+		{
+			overlayManager.add(hydraAttackCounterOverlay);
+		}
+
+		if (prayerOverlay)
+		{
+			overlayManager.add(hydraPrayerOverlay);
+			overlayManager.add(hydraPrayerAttackCounterOverlay);
+		}
+
+		if (poisonOverlay)
+		{
+			overlayManager.add(hydraPoisonOverlay);
+		}
+
+		resetHydras();
+		poisonProjectiles.clear();
+
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		overlayManager.remove(hydraAttackCounterOverlay);
+		overlayManager.remove(hydraPrayerOverlay);
+		overlayManager.remove(hydraPrayerAttackCounterOverlay);
+		overlayManager.remove(hydraPoisonOverlay);
+		resetHydras();
+		poisonProjectiles.clear();
+
+	}
+
+	@Subscribe
+	private void onConfigChanged(ConfigChanged event)
+	{
+		final String configGroupName = "hydra";
+		final String configItemAttackCounter = "attackCounterOverlay";
+		final String configItemPrayerOverlay = "prayerOverlay";
+		final String configItemPoisonProjectileOverlay = "poisonProjectileOverlay";
```

Comment:
As these are static, they should be a private static final variable in the global args

Reply:
Yeah, this makes the compiler able to inline it if it's static. It doesn't trust static instance fields to be the same

Reply:
Ah ok, I'm fairly oblivious to what compiler optimizations can be made. Are there any common ones? (e.g. regarding static/final modifiers)

Reply:
None that I can really think of, the compiler may (probably should) even inline those variable assignments there. When I typed that comment I didn't realize they were vars inside a method, and thought it were fields.